### PR TITLE
NPCs have same restrictions as player for attacking inside containers

### PR DESCRIPTION
**Pull request by Elijahrane**
## About the PR

Fixes half of the issues in https://github.com/space-wizards/space-station-14/issues/13442. Previous behavior was both unfair (NPCs were effectively cheating and could ignore rules you are bound by for attacking people) and confusing (NPCs could attack something in your hand and it would look like they were attacking you but they weren't.)

**Media**

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

### CODE REVIEW

**Comment by Elijahrane** [comment]
Ranged combat intentionally unchanged because the actual result of the attack is identical to if a player did afaict, they can't teleport bullets inside of a backpack or anything

File: Content.Server/NPC/Systems/NPCCombatSystem.Melee.cs [hunk 1]
```diff
@@ -122,6 +122,13 @@ private void Attack(NPCMeleeCombatComponent component, TimeSpan curTime, EntityQ
             return;
         }
 
+        // Players cannot attack entities inside containers since melee context menu was removed, so mirror this for NPCs.
+        if (_containers.IsEntityInContainer(component.Target))
+        {
+            component.Status = CombatStatus.TargetUnreachable;
+            return;
+        }
+
```
Comment (metalgearsloth) [inline Content.Server/NPC/Systems/NPCCombatSystem.Melee.cs:131]: Doesn't this mean players can still attack targets in containers with malicious messages, seems more important to fix that.

Comment (Elijahrane) [reply Content.Server/NPC/Systems/NPCCombatSystem.Melee.cs:131]: Hypothetically yes, I can take a look at that

**Comment by AJCM-git** [comment]
@Elijahrane You coming back to this

**Comment by Elijahrane** [comment]
> @Elijahrane You coming back to this

it's kind of changing the scope of the PR entirely and I have way too much on my plate now with a new job

the current check works fine but if you want it fixed the other way then this can be closed

targeting has a bunch of issues:
https://github.com/space-wizards/space-station-14/issues/13442
https://github.com/space-wizards/space-station-14/issues/11613
(unlisted apparently) wide attacking inside a mech will attack the mech, there's no good passthrough for attacking from inside a container

probably needs a full pass to be done properly which would be a ~200 line pr minimum imo

**Pull request closed without merging**